File: process-transportation/transportation-api/app/models/country_model.py
from typing import Optional, List
from models.mongo_model import PyObjectId, MongoModel
from pydantic import Field


class CountryModel(MongoModel):
    id: Optional[PyObjectId] = Field(default_factory=PyObjectId, alias="_id")
    name: str = Field(...)
    cities: List[PyObjectId] = Field(default_factory=PyObjectId, alias='city_ids')
    capacity: int = Field(...)
    visa_available: bool = Field(...)
    visa_requirements: str = Field(...)
    banned_countries: List[PyObjectId] = Field(default_factory=PyObjectId, alias="banned_countries")

class CountriesModel(MongoModel):
    countries: List[CountryModel] = Field(...)

File: refugee-helper/src/main/java/com/example/workflow/models/CountryModel.java
package com.example.workflow.models;

import java.util.ArrayList;

public class CountryModel {
    private String _id;
    private String name;
    private int capacity;
    private ArrayList<String> city_ids;
    private boolean visa_available;
    private String visa_requirements;
    private ArrayList<String> banned_countries;

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    public ArrayList<String> getCity_ids() {
        return city_ids;
    }

    public void setCity_ids(ArrayList<String> city_ids) {
        this.city_ids = city_ids;
    }

    public boolean isVisa_available() {
        return visa_available;
    }

    public void setVisa_available(boolean visa_available) {
        this.visa_available = visa_available;
    }

    public String getVisa_requirements() {
        return visa_requirements;
    }

    public void setVisa_requirements(String visa_requirements) {
        this.visa_requirements = visa_requirements;
    }

    public ArrayList<String> getBanned_countries() {
        return banned_countries;
    }

    public void setBanned_countries(ArrayList<String> banned_countries) {
        this.banned_countries = banned_countries;
    }
}
